### DISCUSSION TITLE
Modify and Search

### DISCUSSION THREAD

**Discussion by naba1710** [community/community#21464] [Other Feature Feedback, Questions, & Ideas]
<p>Hello. I want to modify and sort but it’s only modifying why is it not searching?</p>
<pre><code class="lang-auto">std_file = open("students.txt", "r") #reading the contents in students.txt file
std_records = std_file.readlines() #readlines() return all lines in the file, as a list where each line is an item in the list object

record = 0
for std_record in std_records:
   print("[",(record+1),"]",std_record.strip()) #printing each line from text file
   record+=1

record_num=int(input("\nEnter record number you wanted to edit: "))
print()
std_details = std_records[record_num-1].split(",") #value of selected record from text file transfer to std_details list variable

record = 0
for std_record in new_list:
   print("[",(record+1),"]",std_record.strip()) #printing each line from text file
   record+=1

element = 0
for record_element in std_details:
   print("[",(element+1),"]",record_element) #printing the selected record details
   element+=1

element_number=int(input("Enter element number you wanted to edit: "))

print("\nCurrent element value:",std_details[element_number-1])

new_element_value = input("Enter new element value: ")

new_line=""

if element_number==4 and record!=record_num:
   new_line="\n" #to be added when updating the record in text file, so each record separated by new line

std_details[element_number-1] = new_element_value+new_line #assigning new value into selected element

updated_record = ','.join(std_details) #join() method takes all items in a list and joins them into one string. updated_record is a string

std_records[record_num-1] = updated_record #now selected line has updated value

new_list = []

while std_records:
   minimum = std_records[0]
   for x in std_records:
      if x &lt; minimum:
         minimum = x
   new_list.append(minimum)
   std_records.remove(minimum)

print()
print("Sorted:")

with open("students.txt", "w") as file:
    file.writelines(std_records) #writing the updated student records into text file

print("\nRecord updated!")

std_file.close()
</code></pre>

**Comment by airtower-luna** [comment]
<p>Hi, it’s hard to help without knowing what exactly is wrong. Like, exactly what output do you get, and what would you expect?</p>
<p>Also the code is hard to read like this, please enclose it in a markdown code block so indentation is preserved, like this:</p>
<pre><code>```py
if a == b:
    print('Hello!')
```
</code></pre>
<p>The <code>py</code> after ``` is optional, it configures Python syntax highlighting. The result should look similar to this:</p>
<pre><code class="lang-auto">if a == b:
    print('Hello!')
</code></pre>

**Comment by naba1710** [comment]
<p>hello, thank you for the respond.<br>
the output i get  is the record of text file and im able to modify it. the output i want is i get the record it should be sorting in ascending order and im able to modify</p>

**Comment by airtower-luna** [comment]
<p>I’m afraid that’s not enough detail. To help we’d need to know exactly (as in, actual content!) how your input file looks, what output you get, and what you want to get.</p>
<p>Also, please edit your first post to use a code block as I described above. Otherwise the code is unreadable.</p>

**Comment by naba1710** [comment]
<p>hello, i have edit it thank you. and that’s how my text file looks like</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/7/b/7be4df71187695cd4a040cd86b492b75ecb43de6.png" data-download-href="/uploads/short-url/hG15mXPoev1S7Ft3oQQOQW4iuRE.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/77396325/181064651-b3f0e343-3a2b-48d6-9b4d-4428eff36ef1.png" alt="image" data-base62-sha1="hG15mXPoev1S7Ft3oQQOQW4iuRE" width="690" height="262" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/7/b/7be4df71187695cd4a040cd86b492b75ecb43de6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">990×377 19.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment, marked as answer]
<p>I think the main issue is that when writing out the results you use <code>std_records</code>, not <code>new_list</code>, which seems to be supposed to hold the sorted data.</p>
<p>That said, I think the code above is not exactly what you use, because <code>new_list</code> is used before being assigned to, which should crash with a <code>NameError</code>.</p>

**Comment by naba1710** [comment]
<p>thank you so much!! i didn’t notice it.</p>
